pkg/expansion/aggregate.go

```diff
@@ -0,0 +1,31 @@
+package expansion
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+)
+
+const childMsgPrefix = "[mock resource created from expanding %s]"
```

Comment:
It seems this prefix is also part of the deny message returned from the admission response. While I understand this is part of the message because it is not the actual resource but the generated mock resource, I do however wonder if this going to create confusion for users who are trying to deploy resources to their cluster as they may not understand how gatekeeper works and why a `mock resource created from expanding` another resource is blocking the deployment. Thoughts? @maxsmythe @davis-haba @sozercan

Reply:
I agree this could be confusing, especially to users who are unfamiliar with Gatekeeper. 

We definitely want to call out the Expansion Template name. That way admins can understand why a constraint that does not match the resource under test is causing a violation, and trace that to the specific Expansion Template.

From the perspective of a user who just wants to deploy a resource, it should be clear that this violation was due to the secondary effect of applying the resource under test. 

It's hard to do this without being overly verbose, but with a well chosen Expansion Template name, a message like `[Implied by deployments-create-pods] you must provide labels...` could satisfy both requirements.

Reply:
+1 to making the message shorter and clearer. I like the `[implied by X]` idea, since it gives the policy admin some control over what message they want to send and adds traceability.

WDYT @ritazh ?

Reply:
is x here the expansion template or the parent resource? In the current implementation, x is the parent resource but it sounds like we want traceability for which expansion template caused the generation? IMO both are good to have but expansion template is not going to make sense to the user who deployed the violating resource.

Reply:
X == expansion template. Current implementation is parent resource, but that doesn't make sense since the parent resource is the object-under-test, which means the user already has that info.

I think the expansion template can make sense if they are named appropriately e.g. `deployments-create-pods` per above. The main thing is to provide this data in a way that can at least be made to be coherent.

IMO either we provide the expansion template name, which puts the onus on the admin to choose good expansion template names, or we don't provide context, which hurts debugability and might lead to odd-sounding errors without the extra context.

Happy to hear better ideas, but none are coming to mind for me.

Reply:
@ritazh Codie is updated to surface the Expansion Template name in the violation message, like we discussed. 

An example, when applying a violating generator:
`
Error from server (Forbidden): error when creating "breadsticks_deployment.yaml": admission webhook "validation.gatekeeper.sh" denied the request: [olivegarden-must-have-breadsticks] [Implied by deployments-create-pods] you must provide labels: {"breadsticks"}
`